src/tools/dataStructures/stack.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const Node = require('./node.js');
+
+class Stack {
+  HEAD = null;
+  constructor(data) {
+    if (data) {
+      if (Array.isArray(data)) {
+        this.fromArray(data);
+      } else if (typeof data !== 'object') {
+        this.push(data);
+      } else {
+        throw new TypeError(
+          'The stack constructor argument must be either array or primitive.' +
+          `\nIts type: ${typeof data}`
+        );
+      }
+    }
+  }
```

Comment:
Maybe it would be better to check whether data is falsey and immediately return if it is:
```suggestion
  constructor(data) {
    if (!data) return;
    if (Array.isArray(data)) {
      this.fromArray(data);
    } else if (typeof data !== 'object') {
      this.push(data);
    } else {
      throw new TypeError(
        'The stack constructor argument must be either array or primitive.' +
        `\nIts type: ${typeof data}`
      );
    }
  }
```